aspects/schema.go

```diff
@@ -138,6 +142,8 @@ func (s *StorageSchema) parse(raw json.RawMessage) (parser, error) {
 		if err := schema.parseConstraints(schemaDef); err != nil {
 			return nil, err
 		}
+	} else if schema.mustConstrain() {
```

Comment:
This works, but at first glance it seems unexpected to have a route that can parse constraints without consulting the 
presumed definitive source indicating if it is required, mustConstrain().

Reply:
That's because while the type may not "require" it (i.e., we can just a have a simple string definition), we still have to parse constraints if they're there. Maybe the name could be improved since "mustConstrain" means "this type requires constraints so there must be a map definition"